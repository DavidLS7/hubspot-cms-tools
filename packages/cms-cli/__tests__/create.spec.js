/*global banjo */

// TODO: Do we have a way to actually go through the full command?
describe('Create command', () => {
  it('works with help positional', async () =>
    expect(await banjo(['create', 'help'])).toMatchSnapshot());

  it('works with help option', async () =>
    expect(await banjo(['create', '--help'])).toMatchSnapshot());

  describe('edge cases', () => {
    it('creates a valid function with a default destination', async () => {});
    it('does not create a function when a directory that does not exist is chosen', async () => {});
    it('does not create a function when a duplicate name is entered', async () => {});
  });

  describe('personalaccesskey', () => {
    describe('function', () => {
      it('creates a valid function with a destination that exists', async () => {});
    });
    describe('module', () => {
      it('creates a valid module with a destination that exists', async () => {});
    });
    describe('template', () => {
      it('creates a valid template with a destination that exists', async () => {});
    });
    describe('website-theme', () => {
      it('creates a valid website-theme with a destination that exists', async () => {});
    });
    describe('react-app', () => {
      it('creates a valid react-app with a destination that exists', async () => {});
    });
    describe('vue-app', () => {
      it('creates a valid vue-app with a destination that exists', async () => {});
    });
    describe('webpack-serverless', () => {
      it('creates a valid webpack-serverless with a destination that exists', async () => {});
    });
  });

  describe('oauth2', () => {
    describe('function', () => {
      it('creates a valid function with a destination that exists', async () => {});
    });
  });
});
