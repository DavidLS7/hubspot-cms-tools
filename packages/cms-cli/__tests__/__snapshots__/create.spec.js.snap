// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create command works with help option 1`] = `
"banjo create <type> <name> [dest]

Create HubSpot CMS assets. Supported assets are function, module, template,
website-theme, react-app, vue-app, and webpack-serverless.

Positionals:
  type  Type of asset
                 [string] [required] [choices: \\"function\\", \\"module\\", \\"template\\",
                  \\"website-theme\\", \\"react-app\\", \\"vue-app\\", \\"webpack-serverless\\"]
  name  Name of new asset                                    [string] [required]
  dest  Destination folder for the new asset, relative to your current working
        directory. If omitted, this argument will default to your current
        working directory.                                              [string]

Options:
  --version    Show version number                                     [boolean]
  --debug, -d  set log level to debug                 [boolean] [default: false]
  --help       Show help                                               [boolean]
"
`;

exports[`Create command works with help positional 1`] = `
"banjo create <type> <name> [dest]

Create HubSpot CMS assets. Supported assets are function, module, template,
website-theme, react-app, vue-app, and webpack-serverless.

Positionals:
  type  Type of asset
                 [string] [required] [choices: \\"function\\", \\"module\\", \\"template\\",
                  \\"website-theme\\", \\"react-app\\", \\"vue-app\\", \\"webpack-serverless\\"]
  name  Name of new asset                                    [string] [required]
  dest  Destination folder for the new asset, relative to your current working
        directory. If omitted, this argument will default to your current
        working directory.                                              [string]

Options:
  --version    Show version number                                     [boolean]
  --debug, -d  set log level to debug                 [boolean] [default: false]
  --help       Show help                                               [boolean]
"
`;
